src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,129 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Class for parsing strings into commands.
+ */
+public class Parser {
+    class EmptyInputException extends RuntimeException {
+        public EmptyInputException() {
+            super("☹ OOPS!!! The description of a todo cannot be empty.");
+        }
+    }
+
+    class InvalidInputException extends RuntimeException {
+        public InvalidInputException() {
+            super("☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
+        }
+    }
+
+    /**
+     * Parses a string into a command.
+     * @param line - the string to parse
+     * @param todolist - the task list to modify
+     * @return 1 if the user wants to quit, 0 otherwise
+     */
+    public int parse(String line, TaskList todolist) {
+        switch (line) {
+        case "":
+            throw new EmptyInputException();
+
+        case "bye":
+            System.out.println("See you again!");
+            return 1;
+
+        case "list":
+            todolist.printData();
+            break;
+
+        case "save":
+            Storage.save(todolist);
+            break;
+
+        default:

```

Comment:
Just a suggestion, but I think the code would look cleaner if there was a separate helper function to check the formatting of the commands (i.e. whether or not the arguments are valid).

And to check the formatting of commands, maybe you might want to consider using the Array.split() command instead of regex. The code might be more readable that way, and there would also be less hard coding (of regex expressions) involved!